adagrams/game.py

```diff
@@ -1,77 +1,72 @@
 from random import randint
 
-LETTER_POOL = {
-    'A': 9, 
-    'B': 2, 
-    'C': 2, 
-    'D': 4, 
-    'E': 12, 
-    'F': 2, 
-    'G': 3, 
-    'H': 2, 
-    'I': 9, 
-    'J': 1, 
-    'K': 1, 
-    'L': 4, 
-    'M': 2, 
-    'N': 6, 
-    'O': 8, 
-    'P': 2, 
-    'Q': 1, 
-    'R': 6, 
-    'S': 4, 
-    'T': 6, 
-    'U': 4, 
-    'V': 2, 
-    'W': 2, 
-    'X': 1, 
-    'Y': 2, 
-    'Z': 1
-}
-
 def draw_letters():
 
-    letter_pool_copy = dict(LETTER_POOL)
-    letter_pool_keys_list = list(letter_pool_copy.keys())
-    letter_pool_length = len(letter_pool_keys_list)
-
-    hand_list = []
-    counter = 0
-    length_of_hand_list = 10
-    
-    while counter < length_of_hand_list:
-
-        number = randint(0, letter_pool_length - 1)
-        letter = letter_pool_keys_list[number]
+    letter_pool = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
 
-        if letter_pool_copy[letter] > 0:
-            hand_list.append(letter)
-            letter_pool_copy[letter] -= 1
-            counter += 1
+    weighted_letter_bag= []
+    for letter, frequency in letter_pool.items():
+        weighted_letter_bag += letter * frequency
 
-    return hand_list
+    hand_of_letters = []
+    MAX_HAND_SIZE = 10
```

Comment:
Placed inside a function, this is a local variable. Even though we want it to be constant, it'd likely be more common to write in lowercase `max_hand_size` when it's inside a function.

Thinking about the code overall, we use this hand limit size of `10` down in `score_word` too when deciding if we should add bonus points. I think there could be value in moving this variable out of the function to be a global constant. If we did move it to the global scope, we would want to keep the all caps naming.